test/unit/int_algolia/scripts/hooks/order/algoliaHooks.test.js

```diff
@@ -0,0 +1,214 @@
+        });
+    });
+
+    test('inventoryUpdate should handle in-store pickup shipment correctly', function () {
+        const Status = require('dw/system/Status');
+        // Arrange
+        const order = mockOrder;
+        
+        // Act
+        const result = algoliaHooks.inventoryUpdate(order);
+        
+        // Assert
+        expect(result).toBeInstanceOf(Status);
+    });
+
+    test('handleInStorePickupShipment should generate correct operations for master-level record model - out of stock - storeAvailability', function () {
+        // Arrange
+        const shipment = mockShipmentInStore;
+        const threshold = 5;
+        const additionalAttributes = ['storeAvailability'];
+        const recordModel = 'master-level';
+        
+        // Act
+        const operations = algoliaHooks.handleInStorePickupShipment(
+            shipment,
+            threshold,
+            additionalAttributes,
+            recordModel
+        );
+        
+        // Assert
+        expect(operations).toMatchSnapshot();
+    });
+
+    test('handleInStorePickupShipment should generate correct operations for variant-level record model - out of stock - storeAvailability', function () {
+        // Arrange
+        const shipment = mockShipmentInStore;
+        const threshold = 5;
+        const additionalAttributes = ['storeAvailability'];
+        const recordModel = 'variant-level';
+        
+        // Act
+        const operations = algoliaHooks.handleInStorePickupShipment(
+            shipment,
+            threshold,
+            additionalAttributes,
+            recordModel
+        );
+        
+        // Assert
+        expect(operations).toMatchSnapshot();
+    });
+
+
+    test('handleStandardShipment should generate correct operations for out of stock products - in_stock - should records be deleted', function () {
+        // Arrange
+        const shipment = mockShipmentStandard;
+        const threshold = 10;
+        const additionalAttributes = ['in_stock'];
+        const recordModel = 'master-level';
+
+        // Act
+        const operations = algoliaHooks.handleStandardShipment(
+            shipment,
+            threshold,
+            additionalAttributes,
+            recordModel
+        );
+
+        // Assert
+        expect(operations).toMatchSnapshot();
+    });
+
+    test('handleStandardShipment should generate correct operations for in-stock products - in_stock', function () {
+        // Arrange
+        const shipment = mockShipmentStandard;
+        const threshold = 1;
+        const additionalAttributes = ['in_stock'];
+        const recordModel = 'master-level';
+        
+        // Act
+        const operations = algoliaHooks.handleStandardShipment(
+            shipment,
+            threshold,
+            additionalAttributes,
+            recordModel
+        );
+        
+        // Assert
+        expect(operations).toMatchSnapshot();
```

Comment:
There is no operations generated when the product is still in stock. So no need for an empty snapshot, you can just assess that there is no operations generated. And update the test description to make it clear that it's what we want.

Reply:
I refactored it and created a more comprehensive approach now.